SmartEVSE-3/src/evse.cpp

```diff
@@ -2999,15 +3021,6 @@ void Timer1S(void * parameter) {
     } // while(1)
 }
 
-void Mongoose_Poll(void * parameter) {
-    while (true) {
-        if (WiFi.isConnected())
-            mg_mgr_poll(&mgr, 1000);
```

Comment:
There is  a good reason mg_mgr_poll isn't in our loop(): it causes a crash.

Coredump:
![image](https://github.com/dingo35/SmartEVSE-3.5/assets/6104159/cc154f89-d8c4-480d-acb2-46fa666b27f0)

Reply:
Each call into the mongoose module needs to happen on the same RTOS thread, as the documentation says: https://mongoose.ws/documentation/tutorials/core/multi-threaded/#overview

I believe that the root cause of the crash you mention is somewhere else and unfortunately, such callback-driven frameworks are notoriously hard to debug. If I find possible causes, I will look for alternative solutions.

In general, preemptive multitasking like FreeRTOS threading is dangerous if the software components are not explicitly protected against it (and most high-level libraries on microcontrollers aren't). I would suggest moving all networking-related stuff on the main task as well as non-time-critical code, i.e. mongoose (including MQTT and the web server), OCPP and Wi-Fi management. Otherwise you always need to bear in mind which task is executing what which makes the code more complex to follow + adds the need for extra synchronization mechanisms.

Reply:
See my email...